Code/Source/RobotImporter/RobotImporterWidget.cpp

```diff
@@ -0,0 +1,106 @@
+                      break;
+                  }
+              })
+    {
+        setWindowTitle(QObject::tr("Robot definition file importer"));
+        QVBoxLayout* mainLayout = new QVBoxLayout(this);
+        mainLayout->setSpacing(20);
+        QLabel* captionLabel = new QLabel(QObject::tr("Select a Unified Robot Description Format (URDF) file to import"), this);
+        captionLabel->setWordWrap(true);
+        mainLayout->addWidget(captionLabel);
+        mainLayout->addWidget(&m_selectFileButton);
+        mainLayout->addWidget(&m_statusLabel);
+        mainLayout->addStretch();
+
+        connect(
+            &m_importerUpdateTimer,
+            &QTimer::timeout,
+            [this]
+            {
+                m_robotImporter.CheckIfAssetsWereLoadedAndCreatePrefab(
+                    [this]()
+                    {
+                        m_importerUpdateTimer.stop();
+                        m_selectFileButton.setEnabled(true);
+                    });
+            });
+
+        QObject::connect(
+            &m_selectFileButton,
+            &QPushButton::clicked,
+            this,
+            [this]()
+            {
+                AZStd::optional<AZStd::string> urdfPath = RobotImporterWidgetUtils::QueryUserForURDFPath(this);
+                if (!urdfPath)
+                {
+                    return;
+                }
+
+                AZ::IO::Path prefabName(AZ::IO::PathView(urdfPath.value()).Filename());
+                prefabName.ReplaceExtension("prefab");
+                AZStd::string prefabDefaultPath(AZ::IO::Path(AZ::Utils::GetProjectPath()) / "Assets" / "Importer" / prefabName);
+                auto prefabPath =
+                    RobotImporterWidgetUtils::ValidatePrefabPathExistenceAndQueryUserForNewIfNecessary(prefabDefaultPath, this);
+                if (!prefabPath)
+                {
+                    ReportError("User cancelled");
+                    return;
+                }
+                m_robotImporter.ParseURDFAndStartLoadingAssets({ urdfPath.value(), prefabPath->c_str() });
+
+                // Disable the button until the import is complete to prevent the user from clicking it again
+                m_selectFileButton.setEnabled(false);
+                // Check whether import is still in progress every 0.5 seconds
+                m_importerUpdateTimer.start(500);
+            });
+        setLayout(mainLayout);
+    }
+
+    void RobotImporterWidget::ReportError(const AZStd::string& errorMessage)
+    {
+        QMessageBox::critical(this, QObject::tr("Error"), QObject::tr(errorMessage.c_str()));
+        m_statusLabel.setText(errorMessage.c_str());
+        AZ_Error("RobotImporterWidget", false, errorMessage.c_str());
+    }
+
+    void RobotImporterWidget::ReportInfo(const AZStd::string& infoMessage)
+    {
+        m_statusLabel.setText(QObject::tr(infoMessage.c_str()));
+        AZ_TracePrintf("RobotImporterWidget", infoMessage.c_str());
```

Comment:
This opens up the possibility of [format string injection](https://owasp.org/www-community/attacks/Format_string_attack) by directly using the input string as the format string.

A simpler method that isn't vulnerable to format string injection is to directly call the `AZ::Debug::Trace::Output` function which doesn't use a format string.
i.e
```c++
AZ::Debug::Trace::Instance().Output("RobotImporterWidget", infoMessage.c_str());
```